src/server/scripts/Kalimdor/zone_orgrimmar.cpp

```diff
@@ -263,8 +264,557 @@ class npc_thrall_warchief : public CreatureScript
+                            olvia->AI()->Talk(SAY_OLVIA_2);
+                        events.ScheduleEvent(EVENT_SCENE_10, 4500);
+                        break;
+                    case EVENT_SCENE_10:
+                        Talk(SAY_RUNTHAK_1);
+                        events.ScheduleEvent(EVENT_SCENE_11, 1500);
+                        break;
+                    case EVENT_SCENE_11:
+                        Talk(SAY_RUNTHAK_2);
+                        events.ScheduleEvent(EVENT_SCENE_12, 4500);
+                        break;
+                    case EVENT_SCENE_12:
+                        Talk(SAY_RUNTHAK_3);
+                        events.ScheduleEvent(EVENT_SCENE_13, 4500);
+                        break;
+                    case EVENT_SCENE_13:
+                        Talk(SAY_RUNTHAK_4);
+                        events.ScheduleEvent(EVENT_RESET, 25000);
+                        break;
+                    case EVENT_RESET:
+                        Reset();
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+        void GetCitizenGuids()
+        {
+            // if one GUID is empty it means all the others are empty as well so we should store them
+            // otherwise do not call for grid search since someone else already activated event once before and guids are stored
+            if (gryshkaGUID.IsEmpty())
+            {
+                std::list<Unit*> citizenList;
+                Trinity::AnyFriendlyUnitInObjectRangeCheck checker(me, me, 25.0f);
+                Trinity::UnitListSearcher<Trinity::AnyFriendlyUnitInObjectRangeCheck> searcher(me, citizenList, checker);
+                me->VisitNearbyObject(20.0f, searcher);
+                for (Unit* target : citizenList)
+                {
+                    switch (target->GetEntry())
+                    {
+                        case NPC_GRYSHKA:
+                            gryshkaGUID = target->GetGUID();
+                            break;
+                        case NPC_OLVIA:
+                            olviaGUID = target->GetGUID();
+                            break;
+                        case NPC_SANA:
+                            sanaGUID = target->GetGUID();
+                            break;
+                        case NPC_FELIKA:
+                            felikaGUID = target->GetGUID();
+                            break;
+                        case NPC_THATHUNG:
+                            thungGUID = target->GetGUID();
+                            break;
+                        case NPC_KAJA:
+                            kajaGUID = target->GetGUID();
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            }
+        }
+
+    private:
+        EventMap events;
+        bool inProgress;
```

Comment:
this field is not currently initialized in the constructor, please initialized it